### PR TITLE
[FEATURE][SERVER] Start session

### PR DESCRIPTION
**Pull request by Drakulix**
Now that the server is able to do something meaningful (at least once #144 is merged), actually start a session.

### CODE REVIEW

File: de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/SarosServer.java [hunk 1]
```diff
@@ -62,6 +63,8 @@ public void start() {
         context.initialize();
```
Comment (srossbach) [inline de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/SarosServer.java]: Can you please create a lifecycle object first ? There is some setup stuff in the context classes that does not belong there and if this code is moved to the lifecylce class then this approach fails badly.

**Comment by Drakulix** [comment]
@srossbach This needs some doc comments, some cleanups and a rebase, but is this what you had in mind? Is there anything obviously wrong (I am not sure, I do understand the whole context stuff yet)?

**Comment by srossbach** [comment]
You SHOULD use this class: https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/context/AbstractContextLifecycle.java to create context.

File: de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/ServerLifecycle.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import de.fu_berlin.inf.dpp.session.ISarosSessionManager;
+import de.fu_berlin.inf.dpp.session.SessionEndReason;
+
+public class ServerLifecycle extends AbstractContextLifecycle {
+
+    private static final Logger log = Logger.getLogger(ServerLifecycle.class);
+    
+    @Override
+    protected Collection<IContextFactory> additionalContextFactories() {
+        List<IContextFactory> factories = new ArrayList<IContextFactory>();
```
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/ServerLifecycle.java]: Code Smell Code Smell: Replace the type specification in this constructor call with the diamond operator ("<>"). ([squid:S2293](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS2293))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=281&issues=AWb__iXssSodgw_KF5fR&open=AWb__iXssSodgw_KF5fR)

[ISSUE_KEY]: AWb__iXssSodgw_KF5fR

File: de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/ServerLifecycle.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    protected void finalizeContext(final ContainerContext context) {
+        context.getComponent(ISarosSessionManager.class).stopSession(
+            SessionEndReason.LOCAL_USER_LEFT);
+
+        context.getComponent(ConnectionHandler.class).disconnect();
+    }
+
+    /*
+    * FIXME This is currently ALPHA, the server always assumes it can connect to
+    * the XMPP server and never gets disconnected. Of course this is unrealistic !
```
Comment (srossbach) [inline de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/ServerLifecycle.java]: You can install some buggish SMACK staff but I would recommend to write your own listener.

File: de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/ServerLifecycle.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    */   
+    private void connectToXMPPServer(final ContainerContext context) {
+        String jidString = ServerConfig.getJID();
+        String password = ServerConfig.getPassword();
+
+        if (jidString == null || password == null) {
+            log.fatal("XMPP credentials are missing! Pass the "
+                + "system properties de.fu_berlin.inf.dpp.server.jid and"
+                + "de.fu_berln.inf.dpp.server.password to the server");
+            System.exit(1);
```
Comment (srossbach) [inline de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/ServerLifecycle.java]: Stopping the context before quitting would be better.

File: de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/ServerLifecycle.java (hunk 2 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/ServerLifecycle.java]: Code Smell Code Smell: Replace the type specification in this constructor call with the diamond operator ("<>"). ([squid:S2293](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS2293))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=281&issues=AWb__iXssSodgw_KF5fR&open=AWb__iXssSodgw_KF5fR)

[ISSUE_KEY]: AWb__iXssSodgw_KF5fR

Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/ServerLifecycle.java]: Code Smell Code Smell: Replace the type specification in this constructor call with the diamond operator ("<>"). ([squid:S2293](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS2293))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=281&issues=AWb__iXssSodgw_KF5fR&open=AWb__iXssSodgw_KF5fR)

[ISSUE_KEY]: AWb__iXssSodgw_KF5fR

**Pull request merged**